lake/api/remote.go

```diff
@@ -129,6 +129,10 @@ func (r *RemoteSession) Delete(ctx context.Context, poolID ksuid.KSUID, branchNa
 	return res.Commit, err
 }
 
+func (r *RemoteSession) DeleteByPredicate(ctx context.Context, poolID ksuid.KSUID, branchName, src string, commit api.CommitMessage) (ksuid.KSUID, error) {
+	return ksuid.Nil, errors.New("unsupported")
```

Comment:
We should implement this, but I guess that can wait.

Reply:
Separate pr.